tests/run_tests.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
```

Comment:
This looks good, thanks for the tests! 

Would you mind converting them to Python? Those can be simple scripts using Plumbum, or tests based on Pytest. It would certainly make this more maintainable in the future... You can run a copier update for the `image-template` and answer `pytest=True` and it will create a Pytest base, if you want, similar to what we already use for other images we maintain.

@Yajo do you think it is worth it for the moment?

Reply:
I totally prefer pytest tests, but @ap-wtioit made a big job already, so let him decide if he wants to go the last mile.

Reply:
I waited to answer that request. Short i would prefer to keep the tests in bash for now.

My reasons are the following. I'm way faster writing such "short" tests in bash and hot fixing stuff on our prod servers also is done in bash. So writing tests in bash for me is always also training to be able to perform server maintenance if it would be ever needed. The suggested Plumbum has a few major (to me) bugs such as these ones: https://github.com/tomerfiliba/plumbum/issues/480 https://github.com/tomerfiliba/plumbum/issues/548 which prevents me from using it effectively for log/output parsing. And refactoring our existing tests with lots of commands and pipes would therefore just be a wrapper to the existing commands. So if we do refactor the tests it seems better to talk to docker with the python api, but that would mean that i / we would need to rewrite all of the test code.

The main reason i didn't answer earlier is that i don't want to prevent you from writing tests because the existing tests are written in bash. So if you think one of you would add tests in the next year if i rewrite our tests to python than i would be willing to rewrite them and check where i can find the time to rewrite them properly.

Reply:
Our choice of running away from bash is based on many problems about it during many years. It's true that it's good for prototyping or issuing commands interactively, but certainly it doesn't grow very well, and its syntax isn't obvious for many cases.

The suggestion about plumbum is not because plumbum is the best tool, but because it's the fastest path to convert a script from bash to python, so it can be easily combined with higher level tools and has a more predictable syntax and workflow.

I don't know if we'll add tests in the next year, but I know that, if we do, they won't be bash.

I appreciate a lot your contribution, and I appreciate it more because it has lots of cpverage. However, if we accept it as is, at some point one of us will have to dedicate a few hours trying to convert this into our coding standards. I feel like it's unfair asking you to do it because you already dedicated to it a lot of time, but I also feel it unfair to have to dedicate that time ourselves because... well, we didn't ask for this after all.

I'm thinking on a middle-ground position that can help us move forward happily. How about this?:

1. @joao-p-marques prepares this project to run tests using our standards: pytest and all of that. You can add a sample test if you want.
2. @ap-wtioit moves these tests inside some pytest test(s), using that test structure. These tests can execute a bash script, effectively wrapping that logic inside pytest.
3. @ap-wtioit tries to make all his following contributions pythonic from the start from now on, to avoid getting tired of us asking the same always 😄

Reply:
That looks like a good option to me :+1: 

I will open another PR preparing the base and let you know once you can rebase here @ap-wtioit 

Thanks again for the contribution and the tests. As @Yajo said, that is already a lot of work and is very appreciated :+1:

Reply:
@joao-p-marques thanks for preparing the test invocation for me, that made it a lot easier.

I've rewritten the tests to use pytest (still invoking docker-compose, but the network setup needed for those tests seemed easier to achive with it than with bare docker / docker api)